lib/driver.rb

```diff
@@ -40,8 +40,39 @@ def add_trip(trip)
       if trip.class != Trip
         raise ArgumentError.new("Can only add trip instance to trip collection")
       end
-
+      @status = :UNAVAILABLE
```

Comment:
I like the idea of setting the status to `:UNAVAILABLE` here. However, because this method is called from `TripDispatcher#load_trips`, this means that only drivers that have _never_ driven a trip will be available. Not quite what you want.

Instead, you might check whether the trip is finished - something like:

```ruby
if trip.end_time.nil?
  @status = :UNAVAILABLE
end
```